### PR TITLE
[REVIEW] Upgrade `arrow-cpp` & `pyarrow` to `7.0.0`

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR upgrades `arrow-cpp` & `pyarrow` to `7.0.0` from `6.0.1`

### CODE REVIEW

**Review by robertmaynard — APPROVED** [review]

**Comment by Ethyling** [comment]
rerun tests

**Comment by Ethyling** [comment]
rerun tests

**Comment by Ethyling** [comment]
rerun tests

File: ci/gpu/java.sh [hunk 1]
```diff
@@ -94,6 +94,9 @@ conda activate rapids
 # https://docs.rapids.ai/maintainers/depmgmt/
 # gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
 # gpuci_mamba_retry install -y "your-pkg=1.0.0"
+gpuci_conda_retry remove --force "arrow-cpp" "pyarrow" "orc" 'arrow-cpp-proc=*=cuda'
+gpuci_mamba_retry install -y "arrow-cpp=7.0.0" "pyarrow=7.0.0" "orc" 'arrow-cpp-proc=*=cuda'
+
```
Comment (galipremsagar) [inline ci/gpu/java.sh:99]:
```suggestion
```

File: ci/cpu/build.sh [hunk 2]
```diff
@@ -64,6 +64,9 @@ conda list --show-channel-urls
 # FIX Added to deal with Anancoda SSL verification issues during conda builds
 conda config --set ssl_verify False
 
+gpuci_conda_retry remove --force "arrow-cpp" "pyarrow" "orc" 'arrow-cpp-proc=*=cuda'
+mamba install -y "arrow-cpp=7.0.0" "pyarrow=7.0.0" "orc" 'arrow-cpp-proc=*=cuda'
+
```
Comment (galipremsagar) [inline ci/cpu/build.sh:69]:
```suggestion
```

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by jakirkham** [comment]
Do we know what caused the failure? Can only spot `ninja: build stopped: subcommand failed.`, which isn't particularly illuminating. Was there a compile error somewhere that I'm missing? If not, is this some kind of hardware oversubscription issue?

**Comment by galipremsagar** [comment]
> Do we know what caused the failure? Can only spot `ninja: build stopped: subcommand failed.`, which isn't particularly illuminating. Was there a compile error somewhere that I'm missing? If not, is this some kind of hardware oversubscription issue?

The compilation errors are due to mix of arrow versions because 22.04 images are being picked up. @ajschmidt8 is working on resolving it.

**Comment by jakirkham** [comment]
rerun tests

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Comment by jakirkham** [comment]
@gpucibot merge

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**